File: abandon-flow-service/src/main/java/io/github/lagom130/lab/client/decoder/ClientErrorDecoder.java
package io.github.lagom130.lab.client.decoder;

import feign.Response;
import feign.codec.ErrorDecoder;
import io.github.lagom130.lab.globalResponse.BizException;

public class ClientErrorDecoder implements ErrorDecoder {
    @Override
    public Exception decode(String s, Response response) {
        return new BizException(response.status(), response.reason());
    }
}

File: delivery-bird-service/src/main/kotlin/io/github/lagom130/lab/client/decoder/ClientErrorDecoder.kt
package io.github.lagom130.lab.client.decoder

import feign.Response
import feign.codec.ErrorDecoder
import io.github.logom130.lab.globalResponse.BizException

class ClientErrorDecoder : ErrorDecoder {
    override fun decode(s: String, response: Response): Exception {
        return BizException(response.status(), response.reason())
    }
}
